### ISSUE TITLE
CakeRequest::header uses empty instead of isset in 2.6

### ISSUE THREAD

**Issue by jtv4k** [cakephp/cakephp#6299]
There are two major issues with the implementation of CakeRequest::header.

```php
public static function header($name) {
	$name = 'HTTP_' . strtoupper(str_replace('-', '_', $name));
	if (!empty($_SERVER[$name])) {
		return $_SERVER[$name];
	}
	return false;
}
```
1. The use of !empty prevents header values of 0 and '', both of which are perfectly valid. Both will always be evaluated as empty and thus return false.
2. There is no distinction between a header not existing and a header with a false value.

A hackable solution is direct key detection with $_SERVER, but that seems like it's going outside the framework.

**Comment by dereuromark** [comment]
I agree that at least `0` could be a valid use case for some headers. Empty string and false might not be very large a difference.
In this case we might have to switch to isset() here as suggested.

**Comment by markstory** [comment]
Empty values fixed in #6300
